### PR TITLE
[eas-cli] update expo packages to SDK 52 versions

### LINKED ISSUE #2696: EAS build: Failed to resolve plugin for module
### Build/Submit details page URL

_No response_

### Summary

Hello Expo Team,

EAS build fails to resolve a plugin inside a library.
This was also an issue in Expo and has been fixed in Expo 52 (more specifically @expo/config-plugins v9)
Links: [here](https://github.com/expo/expo/pull/31569) and [here](https://github.com/expo/expo/commit/4b75917d8456d8852bf95d17add90b464fa5fd62#diff-1de22228381392e614108ecfbf08f4a516e718cd9b736af5a44cbed8a4fd388cR21)

Although the app uses Expo v52 (which depends on config-plugins v9), eas-cli depends on @expo/config-plugins v8 (which seems to be used when doing eas build).
If that's the case, do you have any plans to update?

### Managed or bare?

Bare

### Environment

  expo-env-info 1.2.1 environment info:
    System:
      OS: macOS 15.0
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 20.11.1 - ~/.nvm/versions/node/v20.11.1/bin/node
      Yarn: 1.22.22 - ~/.nvm/versions/node/v20.11.1/bin/yarn
      npm: 10.2.4 - ~/.nvm/versions/node/v20.11.1/bin/npm
      Watchman: 2024.09.09.00 - /opt/homebrew/bin/watchman
    Managers:
      CocoaPods: 1.16.1 - /Users/yevheniismirnov/.rbenv/shims/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 24.1, iOS 18.1, macOS 15.1, tvOS 18.1, visionOS 2.1, watchOS 11.1
      Android SDK:
        API Levels: 28, 30, 31, 32, 33, 34, 35
        Build Tools: 29.0.2, 30.0.2, 30.0.3, 31.0.0, 33.0.0, 33.0.1, 33.0.2, 34.0.0, 35.0.0
        System Images: android-24 | Google APIs ARM 64 v8a, android-25 | Google APIs ARM 64 v8a, android-26 | Google APIs Intel x86 Atom_64, android-27 | Google APIs ARM 64 v8a, android-27 | Google Play Intel x86 Atom, android-28 | Google ARM64-V8a Play ARM 64 v8a, android-29 | Google APIs ARM 64 v8a, android-29 | Google Play ARM 64 v8a, android-30 | ARM 64 v8a, android-30 | Google APIs ARM 64 v8a, android-30 | Google Play ARM 64 v8a, android-31 | Google APIs ARM 64 v8a, android-33 | Google APIs ARM 64 v8a, android-33 | Google Play ARM 64 v8a, android-34 | Google APIs ARM 64 v8a, android-34 | Google Play ARM 64 v8a, android-35 | Google Play ARM 64 v8a
    IDEs:
      Android Studio: 2024.2 AI-242.21829.142.2421.12409432
      Xcode: 16.1/16B40 - /usr/bin/xcodebuild
    npmGlobalPackages:
      eas-cli: 13.2.0
    Expo Workflow: bare

### Error output

Failed to resolve plugin for module "react-native-*" relative to "/home/expo/workingdir/build/packages/sample-expo"

### Reproducible demo or steps to reproduce from a blank project

.

### PR DESCRIPTION
**Pull request by szdziedzic**
# Why

Update `@expo` packages to SDK 52 versions

# How

Update `@expo` packages to SDK 52 versions

Fixes #2696

It seems like the `privacy` field was removed from the expo config in the SDK 52 update: https://expo.dev/changelog/2024/11-12-sdk-52. 

I used `AppPrivay.Unlisted` for the `createApp` mutation input for which this is still required. It was an option selected for an app when `exp.privacy` was undefined.

It also seems that `findWorkspaceRoot` function in `@expo/package-manager` was changed to `resolveWorkspaceRoot`.

# Test Plan

Tests, test commands manually

### CODE REVIEW

**Comment by szdziedzic** [comment]
* **#2706** <a href="https://app.graphite.dev/github/pr/expo/eas-cli/2706?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/expo/eas-cli/2706?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Review by wschurman — COMMENTED** [review]
What is the codepath where this is not using the versioned package for SDK 52? Might be worth focusing on making all versioned packages in eas-cli unused for SDK >= 52/53 like we did for getting the app config.

**Review by szdziedzic — COMMENTED** [review]
>What is the codepath where this is not using the versioned config getter for SDK 52? Might be worth focusing on making all versioned packages in eas-cli unused for SDK >= 52/53 like we did for getting the app config. Upgrading these is always risky (we hit an issue last time we tried to upgrade them IIRC)

Discussed it in DMs with @wschurman and it seems like the best option for now is to go forward with this PR as-is for now then do the decoupling for the SDK 53 release. Decoupling seems like a right long-term call allowing us to avoid many of these upgrade-related issues in the future.

**Review by szdziedzic — COMMENTED** [review]
It seems like this removed `privacy` thing is not a breaking change because `...This field was previously used when the Expo website had optional public-facing pages for projects, which we no longer provide.`

Is it true @wschurman?

**Comment by wschurman** [comment]
https://app.graphite.dev/github/pr/expo/universe/17574/www-No-longer-require-obsolete-privacy-setting-in-app-creation-mutation

File: packages/eas-cli/src/update/configure.ts [hunk 1]
```diff
@@ -149,7 +150,7 @@ async function ensureEASUpdatesIsConfiguredInExpoConfigAsync({
       return {
         projectChanged: true,
         // TODO(cedric): fix return type of `modifyConfigAsync` to avoid `null` for type === success repsonses
```
Comment (wschurman) [inline packages/eas-cli/src/update/configure.ts:152]: yep, this `modifyConfigAsync` is probably a good thing to version next. We versioned `syncUpdatesConfigurationAsync` by creating a CLI for it in expo-updates.

**Review by wschurman — APPROVED** [review]
Let's rebase on that www PR once approved and landed unless getting this out is urgent in which case leaving `AppPrivay.Unlisted` is fine for now (the field is a no-op anyways, doesn't do anything).

**Comment by szdziedzic** [comment]
Thanks!

**Review by brentvatne — APPROVED** [review]

**Pull request merged**